src/main/java/org/sonatype/repository/conan/internal/proxy/ConanProxyFacet.java

```diff
@@ -322,4 +346,19 @@ protected CacheController getCacheController(@Nonnull final Context context) {
     final AssetKind assetKind = context.getAttributes().require(AssetKind.class);
     return checkNotNull(cacheControllerHolder.get(assetKind.getCacheType()));
   }
+
+  @VisibleForTesting
+  static class Config
+  {
+    @JsonDeserialize(using = ConanMatcherDeserializer.class)
+    @JsonTypeInfo(use = Id.NONE)
+    public ConanMatcher conanMatcher;
```

Comment:
I would probably just use a string or enum here rather than go the deserializer route. I do think it's a clever approach but I'd go more with something like we do in our (currently internal) Docker implementation's config and throw an enum there. Even if we stick with the rest of this approach I'd rather make the config just a config and then instantiate the correct matcher explicitly rather than implicitly via a custom deserializer.

Reply:
I have raised this as a separate ticket so that it can be addressed at a later time https://github.com/sonatype-nexus-community/nexus-repository-conan/issues/17